x-pack/legacy/plugins/rollup/public/crud_app/sections/job_create/steps/step_metrics.js

```diff
@@ -47,114 +43,300 @@ const whiteListedMetricByFieldType = {
   },
 };
 
+const checkWhiteListedMetricByFieldType = (fieldType, metricType) => {
+  return !!get(whiteListedMetricByFieldType, [fieldType, metricType]);
+};
+
+// We use an IFFE to associate metricType configs with their
+// associated field types. After processing each of these
+// objects should have a fieldTypes: { date: true, numeric: true }
+// like object.
+const metricTypesConfig = (function () {
+  return [
+    {
+      type: 'avg',
```

Comment:
i wonder if these types should belong in a constants file - that way they could be reused in the tests too. WDYT?

Reply:
Sounds like a plan! Not exactly sure where I'd use them in my spec - will take a look-see.

Reply:
i thought i saw use of `avg`, `max`, etc., in the tests, so was just wondering if it would be useful to extract.